database-service/tests/test_migrations.py

```diff
@@ -1,16 +1,36 @@
-import os
-import sh
+def test_shopozor_structural_migrations_can_be_rolled_back(hasura_client, app_root_folder, db_handler):
+    # Given I've applied structural migrations
+    project_folder = app_root_folder
+    result = hasura_client.apply_migrations(project_folder)
+    assert 0 == result.exit_code
+
+    # When I revert the migrations
+    result = hasura_client.rollback_migrations(project_folder)
+
+    # Then I get no errors
+    assert 0 == result.exit_code
+    # And the database is empty
+    assert db_handler.is_database_empty()
 
 
-def test_migrations_can_be_applied_without_errors(hasura_endpoint):
-    # Given I am in the right location
-    assert os.path.exists('migrations')
-    assert os.path.isfile('config.yaml')
+def test_fixtures_migrations_can_be_rolled_back(hasura_client, app_root_folder, small_fixtures, db_handler, enum_table_names):
+    # Given I've applied fixtures migrations
+    structural_project_folder = app_root_folder
+    hasura_client.apply_migrations(structural_project_folder)
+    hasura_client.apply_migrations(small_fixtures)
+
+    # When I revert the fixtures data
+    result = hasura_client.rollback_migrations(small_fixtures)
+
+    # Then I get no errors
+    assert 0 == result.exit_code
+    # And only the enum tables are not empty
+    assert enum_table_names == db_handler.get_non_empty_tables()
 
-    # When I apply the migrations
-    cmd = sh.Command('hasura')
-    result = cmd('migrate', 'apply', '--endpoint',
-                 hasura_endpoint, '--skip-update-check')
+    # When I revert the structural migrations
+    result = hasura_client.rollback_migrations(structural_project_folder)
 
     # Then I get no errors
     assert 0 == result.exit_code
+    # And the database is empty
+    assert db_handler.is_database_empty()
```

Comment:
This test cleans up after itself, it resets the database to its original state. The problem is that it does not work yet, probably because of a wrong config somewhere. It hangs when it's trying to roll back the structural migrations. I have not yet been able to find out what the problem is but it needs to be fixed.

Reply:
fixed